### DISCUSSION TITLE
GSoC '23 - Creating shareable "albums" from Niffler data sets

### DISCUSSION THREAD

**Discussion by imiro** [Emory-HITI/Niffler#391] [Q&A]
Hi! My name is Aji, I just graduated with a biomedical informatics degree and am new to open-source software development. I am interested in working towards extending Niffler to be able to create shareable "albums". I have matching skillsets and some familiarity with DICOM, though I still need and am eager to learn much more. 

I would like to confirm if I understand the goal of this topic correctly. In my understanding, Niffler will store the extracted image on a local storage and its metadata to a store (mongoDB database for meta-extraction or csv file for cold-extraction). By creating a shareable album, does it mean all images on these stores or only a subset? If it's a subset, what is the current method people use to "filter" these datasets and choose which image to share?

Also, one of the suggested approaches is to use Kheops. I don't have prior experience with it but I have started researching by going through Kheops documentation and found that it has an API to manage albums (create, modify, delete). Therefore, one approach I can think of is to utilize these APIs. I am thinking of a new Niffler module that could:

- authenticates a user to a Kheops instance, 
- allow them to select a subset of data extracted from the other modules (if required), 
- make API calls to Kheops to create a shareable album. 
 
This is assuming that both Niffler and Kheops are configured to connect to the same PACS system. Does this approach make sense? Feedback would be much appreciated. Thank you!

**Comment by pradeeban** [comment]
I was thinking more along the lines, Kheops connected to the local file system where Niffler is storing the DICOM files it retrieved from the PACS. i.e., Kheops connecting to the DestAET rather than the SrcAET (which is the PACS).

Other points are fine.

**Reply by imiro** [reply]
I see, thank you for clarifying @pradeeban. May I confirm with you a couple more things:

- From the currently existing modules, which one(s) do this album sharing functionality is desired to work together with? I'm assuming it's the cold-extraction module but I might be wrong.
- What kind of interface is preferred to allow user to select subset of data? I imagine that if the Kheops instance is connected to the _DestAET_, then user will be able to directly use Kheops to create an album and select which studies to include. I don't understand yet what new functionality is desired on this front.

Thank you!

**Reply by pradeeban** [reply]
>> From the currently existing modules, which one(s) do this album sharing functionality is desired to work together with? I'm assuming it's the cold-extraction module but I might be wrong.

Your assumption is correct.

>> What kind of interface is preferred to allow user to select subset of data? I imagine that if the Kheops instance is connected to the DestAET, then user will be able to directly use Kheops to create an album and select which studies to include. I don't understand yet what new functionality is desired on this front.

Yes, this is correct. We just need seamless integration. It is a "175 Hours" project that is rated "Easy" on the difficulty scale. But at the same time, it is not merely deploying Kheops. We certainly need to make this creating albums feature a "first-class citizen" of Niffler, perhaps even a separate new module. How can we create and share datasets around from the DestAET.

**Reply by imiro** [reply]
Thank you @pradeeban and sorry for the late reply. I am trying to construct a proposal and try to submit it soon. Hopefully I can craft something quickly as a proof of concept and include it in the proposal.
